stackslib/src/net/download/nakamoto/tenure_downloader_set.rs

```diff
@@ -273,10 +325,11 @@ impl NakamotoTenureDownloaderSet {
     /// this up with a call to `clear_available_peers()`.
     pub fn clear_finished_downloaders(&mut self) {
         for downloader_opt in self.downloaders.iter_mut() {
-            let Some(downloader) = downloader_opt else {
-                continue;
-            };
-            if downloader.is_done() {
+            if downloader_opt
+                .as_ref()
+                .map(|dl| dl.is_done())
+                .unwrap_or(false)
+            {
                 *downloader_opt = None;
             }
```

Comment:
This refactoring seems less clear to me. How about:
```rust
        for downloader_opt in &mut self.downloaders {
            if let Some(dl) = downloader_opt {
                if dl.is_done() {
                    *downloader_opt = None;
                }
            }
        }
```

Reply:
I prefer to minimize indentations.  Can I just add a code comment?